packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -59,14 +60,37 @@ export function hasOne(definition?: Object) {
   return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
 }
 
-/**
- * Decorator for hasMany
- * @param definition
- * @returns {(target:any, key:string)}
- */
-export function hasMany(definition?: Object) {
-  const rel = Object.assign({type: RelationType.hasMany}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function hasMany(definition?: Partial<HasManyDefinition>) {
+  return function(target: Object, key: string) {
+    const propMeta = MetadataInspector.getPropertyMetadata(
+      MODEL_PROPERTIES_KEY,
+      target,
+      key,
+    );
+    const allPropMeta = MetadataInspector.getAllPropertyMetadata(
+      MODEL_PROPERTIES_KEY,
+      target,
+    )!;
+    let meta = {};
+    for (const property in allPropMeta) {
+      if (allPropMeta[property].id === true) {
+        Object.assign(meta, {keyFrom: property}, definition);
+      }
+    }
```

Comment:
We should NOT calculate the PK as part of the decoration as it assumes that:

1. `@property` is applied before `@hasMany`
2. All `@property` decorations have been applied before `@hasMany`

None of them is true.

Reply:
Yes, I think you and @shimks have discussed storing the relation definition at the same level of the model definition and to possibly get the PK info from there IIUC?

Reply:
Feedback addressed by having relation metadata be built when `@model` is being resolved